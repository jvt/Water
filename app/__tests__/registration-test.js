import {
    TextInput,
    Image,
    Text
} from 'react-native';
import Button from 'apsl-react-native-button';
import React from 'react';
import Registration from '../registration';
import Renderer from 'react-test-renderer';
import SegmentedControlTab from 'react-native-segmented-control-tab';
import BusyIndicator from 'react-native-busy-indicator';

describe('Registration', () => {
    test('renders correctly', ()  => {
        const login = Renderer.create(
            <Registration />
        ).toJSON();
        expect(login).toMatchSnapshot();
    });

    test('WaterImage', () => {
    	const instance = Renderer.create(
    		<Image
    		Style={{}}
    		Droplet
    		/>
    	);

    	expect(instance.toJSON()).toMatchSnapshot();
    })

    test('RegisterText', ()  => {
        const instance = Renderer.create(
            <Text
            Style={{}}>
            Register
            </Text>
        );

        expect(instance.toJSON()).toMatchSnapshot();
    });

    test('UsernameField', ()  => {
        const instance = Renderer.create(
            <TextInput
            Style={{}}
            Username
            />
        );

        expect(instance.toJSON()).toMatchSnapshot();
    });

    test('PasswordField', () => {
        const instance = Renderer.create(
            <TextInput
            Style={{}}
            Password
            />
        );

        expect(instance.toJSON()).toMatchSnapshot();
    });

    test('ConfirmPasswordField', () => {
        const instance = Renderer.create(
            <TextInput
            Style={{}}
            Confirm Password
            />
        );

        expect(instance.toJSON()).toMatchSnapshot();
    });

    test('RoleTabs', ()  => {
        const instance = Renderer.create(
            <SegmentedControlTab
            values={['User', 'Worker', 'Manager', 'Admin']}
			selectedIndex={0}
            Style={{}}
            Role
            />
        );

        expect(instance.toJSON()).toMatchSnapshot();
    });

    test('RegisterButton', () => {
        const instance = Renderer.create(
            <Button
            Style={{}}>
            Register
            </Button>
        );

        expect(instance.toJSON()).toMatchSnapshot();
    });

    test('ReturnButton', () => {
        const instance = Renderer.create(
            <Button
            Style={{}}>
            Return
            </Button>
        );

        expect(instance.toJSON()).toMatchSnapshot();
    });

    test('BusyIndicator', () =>  {
        const instance = Renderer.create(
            <BusyIndicator />
        );

        expect(instance.toJSON()).toMatchSnapshot();
    });
});
