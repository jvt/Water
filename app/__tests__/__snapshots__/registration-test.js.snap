exports[`Registration BusyIndicator 1`] = `<View />`;

exports[`Registration ConfirmPasswordField 1`] = `
<TextInput
  Confirm={true}
  Password={true}
  Style={Object {}} />
`;

exports[`Registration PasswordField 1`] = `
<TextInput
  Password={true}
  Style={Object {}} />
`;

exports[`Registration RegisterButton 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "fontSize": 18,
          "textAlign": "center",
        },
        undefined,
      ]
    }>
    Register
  </Text>
</View>
`;

exports[`Registration RegisterText 1`] = `
<Text
  Style={Object {}}
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail">
  Register
</Text>
`;

exports[`Registration ReturnButton 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "fontSize": 18,
          "textAlign": "center",
        },
        undefined,
      ]
    }>
    Return
  </Text>
</View>
`;

exports[`Registration RoleTabs 1`] = `
<View
  removeClippedSubviews={true}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flexDirection": "row",
      },
      Object {},
    ]
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0076FF",
        "borderBottomLeftRadius": 5,
        "borderColor": "#0076FF",
        "borderTopLeftRadius": 5,
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          undefined,
          Object {
            "color": "white",
          },
          undefined,
        ]
      }>
      User
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#0076FF",
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          undefined,
          Object {},
          Object {},
        ]
      }>
      Worker
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#0076FF",
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          undefined,
          Object {},
          Object {},
        ]
      }>
      Manager
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomRightRadius": 5,
        "borderColor": "#0076FF",
        "borderTopRightRadius": 5,
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          undefined,
          Object {},
          Object {},
        ]
      }>
      Admin
    </Text>
  </View>
</View>
`;

exports[`Registration UsernameField 1`] = `
<TextInput
  Style={Object {}}
  Username={true} />
`;

exports[`Registration WaterImage 1`] = `
<Image
  Droplet={true}
  Style={Object {}} />
`;

exports[`Registration renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "#007bff",
        "borderBottomWidth": 1,
        "borderColor": "#006acd",
        "bottom": 0,
        "left": 0,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": Array [
          Object {
            "translateY": -1334,
          },
        ],
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "flexDirection": "row",
            "padding": 10,
          }
        }>
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "marginLeft": 10,
            }
          } />
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }>
        <Image
          source={1}
          style={
            Object {
              "height": 175,
              "width": 80,
            }
          } />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "200",
            "textAlign": "center",
          }
        }>
        Register
      </Text>
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Username"
        returnKeyType="next"
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 50,
            "padding": 10,
          }
        } />
      <TextInput
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Password"
        returnKeyType="go"
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 20,
            "padding": 10,
          }
        } />
      <TextInput
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Confirm Password"
        returnKeyType="go"
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 20,
            "padding": 10,
          }
        } />
      <View
        style={
          Object {
            "marginBottom": -20,
            "padding": 20,
            "paddingBottom": 0,
          }
        }>
        <View
          removeClippedSubviews={true}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "flexDirection": "row",
              },
              Object {},
            ]
          }>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#0076FF",
                "borderBottomLeftRadius": 5,
                "borderColor": "#0076FF",
                "borderTopLeftRadius": 5,
                "borderWidth": 1,
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
                "paddingVertical": 5,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  undefined,
                  Object {
                    "color": "white",
                  },
                  undefined,
                ]
              }>
              User
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderColor": "#0076FF",
                "borderWidth": 1,
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
                "paddingVertical": 5,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  undefined,
                  Object {},
                  Object {},
                ]
              }>
              Worker
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderColor": "#0076FF",
                "borderWidth": 1,
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
                "paddingVertical": 5,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  undefined,
                  Object {},
                  Object {},
                ]
              }>
              Manager
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomRightRadius": 5,
                "borderColor": "#0076FF",
                "borderTopRightRadius": 5,
                "borderWidth": 1,
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
                "paddingVertical": 5,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  undefined,
                  Object {},
                  Object {},
                ]
              }>
              Admin
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "rgba(65, 163, 221, 1)",
            "borderRadius": 8,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 50,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
                "fontSize": 18,
                "textAlign": "center",
              },
              Object {
                "fontSize": 18,
              },
            ]
          }>
          Register
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "borderRadius": 8,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
                "fontSize": 18,
                "textAlign": "center",
              },
              Object {
                "fontSize": 18,
              },
            ]
          }>
          Return
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View />
</View>
`;
